### PR TITLE
[Hotfix] Crash when stats loss is used

### PR DESCRIPTION
**Pull request by kacpnowak**
## Description

Fixes three bugs:
- crash when `stats` loss is used
- crash when `secrets` key is not present in private config
- crash when worker's ID is 0

## Type of Change

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation update

## Issue Number

Closes https://github.com/ecmwf/WeatherGenerator/issues/307

## Code Compatibility

-   [x] I have performed a self-review of my code

### Code Performance and Testing

-   [x] I ran the `uv run train` and (if necessary) `uv run evaluate` on a least one GPU node and it works 
-   [ ] If the new feature introduces modifications at the config level, I have made sure to have notified the other software developers through Mattermost and updated the paths in the `$WEATHER_GENERATOR_PRIVATE` directory

### Dependencies

-   [ ] I have ensured that the code is still pip-installable after the changes and runs
-   [ ] I have tested that new dependencies themselves are pip-installable.
-   [ ] I have not introduced new dependencies in the inference portion of the pipeline

### Documentation

-   [ ] My code follows the style guidelines of this project
-   [ ] I have updated the documentation and docstrings to reflect the changes
-   [ ] I have added comments to my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/weathergen/utils/config.py [hunk 1]
```diff
@@ -241,7 +241,10 @@ def _load_private_conf(private_home: Path | None) -> DictConfig:
     private_cf["model_path"] = (
         private_cf["model_path"] if "model_path" in private_cf.keys() else "./models"
     )
-    del private_cf["secrets"]
+
+    if "secrets" in private_cf:
+        del private_cf["secrets"]
```
Comment (tjhunter) [inline src/weathergen/utils/config.py:246]: I missed it. We should not use `del`. 

`private_cf.pop("secrets", None)` is all you need

File: src/weathergen/train/trainer.py [hunk 2]
```diff
@@ -451,15 +455,17 @@ def compute_loss(
...
-            None
-            if not log_data
-            else [
-                preds_all,
-                targets_all,
-                targets_coords_raw_rt,
-                targets_times_raw_rt,
-                targets_lens,
-            ],
+            (
```
Comment (tjhunter) [inline src/weathergen/train/trainer.py:458]: strange that the formatter gives different results here.

**Review by tjhunter — APPROVED** [review]
one style comment. looks good.

File: src/weathergen/datasets/tokenizer_masking.py [hunk 3]
```diff
@@ -122,7 +122,7 @@ def __init__(self, healpix_level: int):
         )
 
         worker_info = torch.utils.data.get_worker_info()
-        div_factor = worker_info.id if worker_info is not None else 1
+        div_factor = (worker_info.id + 1) if worker_info is not None else 1
```
Comment (clessig) [inline src/weathergen/datasets/tokenizer_masking.py:125]: Did this actually get triggered for you? It really shouldn't

File: src/weathergen/train/trainer.py (hunk 2 above)
Comment (clessig) [reply src/weathergen/train/trainer.py:458]: We had this before with Kacper. Any idea what might trigger this behavior?

**Review by clessig — APPROVED** [review]

File: src/weathergen/datasets/tokenizer_masking.py (hunk 3 above)
Comment (kacpnowak) [reply src/weathergen/datasets/tokenizer_masking.py:125]: It doesn't get triggered but I changed it just in case

File: src/weathergen/train/trainer.py (hunk 2 above)
Comment (kacpnowak) [reply src/weathergen/train/trainer.py:458]: Tim pointed out that I'm using Python 3.12.0 instead of 3.12.9

Comment (clessig) [reply src/weathergen/train/trainer.py:458]: @tjhunter  Should we fix the python version more tightly then?

https://github.com/ecmwf/WeatherGenerator/blob/f0a28bd35089ad5ab570181cfe8dea008e64e8ef/pyproject.toml#L10

File: src/weathergen/datasets/tokenizer_masking.py (hunk 3 above)
Comment (clessig) [reply src/weathergen/datasets/tokenizer_masking.py:125]: Needs to be fixed, but just wanted to make sure there were no side effects that I was not aware of / understanding.

Good exercise: why didn't it get triggered?
